src/main/java/com/amplitude/api/AmplitudeClient.java

```diff
@@ -1161,6 +1174,9 @@ protected long logEvent(String eventType, JSONObject eventProperties, JSONObject
             if (appliedTrackingOptions.shouldTrackAdid() && deviceInfo.getAdvertisingId() != null) {
                 apiProperties.put("androidADID", deviceInfo.getAdvertisingId());
             }
+            if (appliedTrackingOptions.shouldTrackAppSetId() && deviceInfo.getAppSetId() != null) {
```

Comment:
deviceInfo.getAppSetId() can be set as a variable outside and earlier. For code readability (doesn't actually affect the compiler)

Reply:
@dantetam Not quite get `deviceInfo.getAppSetId() can be set as a variable outside and earlier.` Do you mean move it as a variable and define it earlier inside this function?

Reply:
String appSetId = deviceInfo.getAppSetId();

Reply:
@dantetam I see your point now, one thing is it may make sense if `appliedTrackingOptions.shouldTrackAppSetId()` is true, then the variable can be used, but if that't not then we don't need a variable for it and it won't be used later in somewhere. If the two if clause order is reverse then it makes more sense as we would always check the app set id. For the code block here, change it or not won't cause too much difference, just some thought for it.